flink-connector-aws/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumerTest.java

```diff
@@ -284,7 +284,7 @@ public void testListStateChangedAfterSnapshotState() throws Exception {
                 new FlinkKinesisConsumer<>("fakeStream", new SimpleStringSchema(), config);
         FlinkKinesisConsumer<?> mockedConsumer = spy(consumer);
 
-        RuntimeContext context = new MockStreamingRuntimeContext(true, 1, 1);
+        RuntimeContext context = new MockStreamingRuntimeContext(true, 1, 0);
```

Comment:
Can you provide some context on why this is needed?

Reply:
Sure, in 1.19 this test is failing with:
```
java.lang.IllegalArgumentException: Task index must be less than parallelism.
	at org.apache.flink.util.Preconditions.checkArgument(Preconditions.java:138)
	at org.apache.flink.api.common.TaskInfoImpl.<init>(TaskInfoImpl.java:68)
	at org.apache.flink.api.common.TaskInfoImpl.<init>(TaskInfoImpl.java:44)
```
Failed check: https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/api/common/TaskInfoImpl.java#L68-L70